terraform/azure/modules/0_landing_zone/agw-public/main.tf

```diff
@@ -0,0 +1,284 @@
+      tags["center"],
+      tags["environment"],
+      tags["escid"],
+      tags["funding_source"],
+      tags["pii_data"],
+      tags["security_compliance"],
+      tags["security_steward"],
+      tags["support_group"],
+      tags["system"],
+      tags["technical_poc"],
+      tags["technical_steward"],
+      tags["zone"]
+      ]
+    create_before_destroy = true
+    }
+}
+
+# Create Managed Identity to allow AGW to read Certificate from KeyVault
+resource "azurerm_key_vault_access_policy" "agw_mi_policy" {
+  depends_on         = [azurerm_user_assigned_identity.agw_mi]
+  key_vault_id       = data.azurerm_key_vault.key_vault.id
+  tenant_id          = data.azurerm_client_config.current.tenant_id
+  object_id          = azurerm_user_assigned_identity.agw_mi.principal_id
+  secret_permissions = ["Get","List"]
+}
+
+# Create WAF Policy. By default should be blocked on NSG
+# resource "azurerm_web_application_firewall_policy" "agw_waf_policy" {
+#   name                = "${var.resource_prefix}-waf-policy"
+#   resource_group_name = data.azurerm_resource_group.rg.name
+#   location            = data.azurerm_resource_group.rg.location
+
+#   managed_rules {
+#     managed_rule_set {
+#       type    = "OWASP"
+#       version = "3.2"
+#     }
+#   }
+
+#   policy_settings {
+#     enabled                     = true
+#     mode                        = "Detection"
+#     request_body_check          = true
+#     file_upload_limit_in_mb     = 100
+#     max_request_body_size_in_kb = 128
+#   }
+
+#   custom_rules {
+#     name      = "Rule1"
+#     priority  = 1
+#     rule_type = "MatchRule"
+
+#     match_conditions {
+#       match_variables {
+#         variable_name = "RemoteAddr"
+#       }
+
+#       operator           = "IPMatch"
+#       negation_condition = false
+#       match_values       = ["10.16.3.0/24", "10.16.0.128/26","10.16.2.0/27"]
+#     }
+
+#     action = "Allow"
+#   }
+  
+#   lifecycle {
+#     ignore_changes = [ 
+#       tags["business_steward"],
+#       tags["center"],
+#       tags["environment"],
+#       tags["escid"],
+#       tags["funding_source"],
+#       tags["pii_data"],
+#       tags["security_compliance"],
+#       tags["security_steward"],
+#       tags["support_group"],
+#       tags["system"],
+#       tags["technical_poc"],
+#       tags["technical_steward"],
+#       tags["zone"]
+#       ]
+#     create_before_destroy = true
+#     }
+# }
+
+# Configure Public App Gateway
+resource "azurerm_application_gateway" "agw_public" {
+  name                = "${var.resource_prefix}-agw-public"
+  depends_on          = [azurerm_public_ip.agw_public_ip,azurerm_key_vault_access_policy.agw_mi_policy,azurerm_user_assigned_identity.agw_mi]
+  resource_group_name = data.azurerm_resource_group.rg.name
+  location            = data.azurerm_resource_group.rg.location
+  # Uncomment if WAF Policy is Required
+  # firewall_policy_id  = azurerm_web_application_firewall_policy.agw_waf_policy.id
```

Comment:
Not sure if this is intended to be used. But if it is a default value of null for the variable will be better here.